### ISSUE TITLE
Statelessness test: Does the CNF use a non-cloud native data store: local volumes on the node?

### ISSUE THREAD

**Issue by lixuna** [cncf/cnf-conformance#269]
[Acceptance Criteria](https://github.com/cncf/cnf-conformance/issues/269#issuecomment-645545163)
---

### Statelessness test: Does the CNF use a non-cloud native data store: local volumes on the node?

**Short Description:** 
- Check for non-cloud native data stores
- Does the CNF use local block volumes on the node?

Reference:
- https://kubernetes.io/blog/2019/04/04/kubernetes-1.14-local-persistent-volumes-ga
- https://kubernetes.io/docs/concepts/storage/volumes/#local

Raw block volumes to be explored in another test:
- https://kubernetes.io/docs/concepts/storage/volumes/#csi-raw-block-volume-support
- https://kubernetes.io/docs/concepts/storage/persistent-volumes/#raw-block-volume-support

**Test Category:** 
- Statelessness (eg. Cloud native State) 

**Type of test (static or runtime)**
- Runtime

**Scoring:** 
- tbd

**Are there any pre-requisites?**
- tbd

**Command name:**
- tbd

---

### Implementation Tasks: TBD

**Environment set up tasks:**
- [ ]  if needed

**Upstream tool set up tasks: (conformance suite + upstream tools)**
- [ ]  if needed

**Code implementation tasks:**
- [ ]  update pre-req checker - N/A

**Documentation tasks:**
- [ ] Update [Test Categories md](https://github.com/cncf/cnf-conformance/blob/master/TEST-CATEGORIES.md) if needed 
- [ ] Update [USAGE md](https://github.com/cncf/cnf-conformance/blob/master/USAGE.md) if needed 
- [ ] Update [installation instructions](https://github.com/cncf/cnf-conformance#installation)  if needed 

### QA tasks

Dev Review: 
- [ ] walk through A/C
- [ ] do you get the expected result?
- [ ] if yes, 
   - [ ] move to `Needs Peer Review` column  
   - [ ] create Pull Request and follow check list
   - [ ] Assign 1 or more people for peer review
- [ ] if no, document what additional tasks will be needed

Peer review:
- [ ] walk through A/C
- [ ] do you get the expected result?
- [ ] if yes, 
   - [ ] move to `Reviewer Approved` column
   - [ ] Approve pull request
- [ ] if no, 
   - [ ] document what did not go as expected, including error messages and screenshots (if possible)
   - [ ] Add comment to pull request
   - [ ] request changes to pull request

**Comment by taylor** [comment]
DRAFT Acceptance Criteria: building on A/C from #159 
---

### Acceptance Criteria for peer review:
--
### Documentation is updated:
- [ ]  Documentation lists new test name
- [ ]  Test Categories md is updated
- [ ]  USAGE md is updated
- [ ]  Installation instructions is updated

###  Common steps
- [ ]  Find or create a K8s cluster to use
   - Talk with team for access to an existing cluster
   - [Kind K8s clusters](https://github.com/cncf/cnf-conformance/blob/master/KIND-INSTALL.md) will work as well
- [ ]  Set the KUBECONFIG environment variable for accessing the target K8s cluster
- [ ]  As a developer, Pull down latest version of CNF conformance test code from {FEATURE} branch
     - [ ]  Check that {FEATURE} Branch passes all A/C 
     - [ ]  Check that {FEATURE} Branch passes tests in Travis CI
- [ ]  As a peer reviewer, check that PR passes Travis CI tests
     - [ ]  if yes, merge PR to master
     - [ ]  if no, make a comment and request that developer make changes to pass tests 
- [ ]  As a peer reviewer, Pull down latest version of CNF conformance test code from `master` branch
- [ ]  Run `shards install`

### Spec test should pass
- [ ]  Run `crystal spec`
- [ ]  I expect to see PASSED for the tests


### Update Prerequistite checker
- N/A

### Update example CNFs cnf-conformance.yml, if needed
- [ ] continue, If there is an update to cnf-conformance.yml
- [ ] for each CNF Example
- [ ] Update the example CNF's cnf-conformance.yml to include the new changes
- [ ] Update the example CNF's README if needed
- [ ] Test the example CNF following it's README

### Update yml validator, if needed
- N/A

### When using the test `cnf-conformance volume_localstorage_not_found` command the sample "good" CNF should PASS test successfully:
- [ ] Run `./cnf-conformance setup`
- [ ] Run  `./cnf-conformance sample_coredns_setup`
- [ ] Run  `./cnf-conformance volume_localstorage_not_found`
- [ ] I expect to see the output and results from the test
- [ ] I expect to see PASSED for the test. Example:
         - [ ] "✔️ PASSED: hostPath volumes not found [EMOJI]"

### When using the test `crystal spec --tag `volume_localstorage_not_found`` command the sample "good" CNF should PASS and the "bad example CNF"  should FAIL test:
- [ ] Run `crystal spec --tag volume_localstorage_not_found`
- [ ] I expect to see the output and results from the test
- [ ] I expect to see "✔️ PASSED: local storage volumes not found [EMOJI]
- [ ] I expect to see "❌ FAIL: local storage volumes found [EMOJI]"

### I would expect the CNF to pass when I run the entire test category suite (`statelessness`)
- [ ]  Follow common steps above
- [ ]  Run  `./cnf-conformance sample_coredns_setup`
- [ ]  Run  `./cnf-conformance statelessness`
- [ ]  I expect to see the output and results from all tests in the test category
- [ ]  I expect to see PASSED for the test and test category 

### I would expect the CNF to pass when I run the entire test suite (`all`)
- [ ]  Follow common steps above
- [ ]  Run  `./cnf-conformance sample_coredns_setup`
- [ ]  Run  `./cnf-conformance all`
- [ ]  I expect to see the output from the test
- [ ]  I expect to see PASSED for the test

### Screenshots of new test and results in the comments
- [ ]  New test should be added to the points file
- [ ]  New test score should be added to the final score
- [ ] I can see a screenshot of the new test in the comments of this ticket

**Comment by taylor** [comment]
For emoji ideas see https://github.com/cncf/cnf-conformance/issues/159#issuecomment-645528848

**Comment by taylor** [comment]
Reference:
- https://kubernetes.io/blog/2019/04/04/kubernetes-1.14-local-persistent-volumes-ga
- https://kubernetes.io/docs/concepts/storage/volumes/#local
- https://kubernetes.io/docs/concepts/storage/persistent-volumes/#raw-block-volume-support

**Comment by taylor** [comment]
Level of effort in points: 0, 1, 2, 3, 5, 8, 13, 21 = 5pts

- @williscool 5
- @agentpoyo 5
- @denverwilliams 5
- @nupejosh 5
- @wavell 5
- @taylor 8
- @lixuna 5

**Comment by agentpoyo** [comment]
![269_no_local_volumes_passed_1](https://user-images.githubusercontent.com/14348948/99435761-33acb100-28d6-11eb-8f41-eb0d10735e93.png)

**Comment by agentpoyo** [comment]
![269_no_local_volumes_passed_2](https://user-images.githubusercontent.com/14348948/99437235-cd289280-28d7-11eb-8133-913768765487.png)

**Comment by agentpoyo** [comment]
When running the workload test in the final A/C steps after downloading the cnf-conformance.yml with wget, it seems to go into a repeated error seen in screenshot below:

![269_no_local_volumes_passed_3](https://user-images.githubusercontent.com/14348948/99437973-c8b0a980-28d8-11eb-9ce8-9f829d0b95cc.png)

Here is the cnf-conformance.yml downloaded from https://raw.githubusercontent.com/cncf/cnf-conformance/master/example-cnfs/coredns/cnf-conformance.yml :

```
$ cat cnf-conformance.yml
---
helm_directory: cnfs/coredns/helm_chart/coredns
# helm_directory: helm_chart
git_clone_url:
install_script:
release_name: coredns
deployment_name: coredns-coredns
deployment_label: k8s-app
service_name: coredns-coredns
application_deployment_names: [coredns-coredns]
docker_repository: coredns/coredns
helm_repository:
  name: stable
  repo_url: https://cncf.gitlab.io/stable
helm_chart: stable/coredns
helm_chart_container_name: coredns
rolling_update_tag: 1.6.7
white_list_helm_chart_container_names: [falco, node-cache, nginx, coredns, calico-node, kube-proxy, nginx-proxy, kube-multus]
```

@wavell @denverwilliams

**Comment by agentpoyo** [comment]
![269_no_local_volumes_passed_4](https://user-images.githubusercontent.com/14348948/99460458-847fd200-28f5-11eb-8029-eb9f0f06fde8.png)

**Comment by lixuna** [comment]
closing issue, test works as expected, errors above were resolved in #474
